src/arch/ucx/machine.C

```diff
@@ -610,6 +614,9 @@ void LrtsAdvanceCommunication(int whileidle)
 #if CMK_SMP
        cnt += ProcessTxQueue();
 #endif
+       if (ucxCtx.numFreeRxReqs >= ucxCtx.rxReqsBatch) {
+           UcxPostRxBuffers();
+       }
```

Comment:
What's the idea behind posting in batches (default being 16/4 = 4)? 

Was it only done to avoid the recursive scenario from before or is there any other advantage?

Reply:
the main idea is to avoid recursion. Batches are used to reduce overhead, so we will repost a bunch of requests rather than posting them separately every time some request is completed